### PR TITLE
Delete code under FEATURE_PREJIT

### LINKED ISSUE #57660: Remove code guarded by FEATURE_PREJIT
This code is now dead with the removal of crossgen1.

### PR DESCRIPTION
**Pull request by elinor-fung**
Fixes #57660

Contributes to #54129 and #53007

cc: @dotnet/crossgen-contrib @AaronRobinsonMSFT @jkoritzinsky @jkotas @agocke

### CODE REVIEW

**Review by jkoritzinsky — APPROVED** [review]

File: src/coreclr/debug/daccess/dacdbiimpl.cpp [hunk 1]
```diff
@@ -1256,32 +1250,9 @@ bool DacDbiInterfaceImpl::GetILImageInfoFromNgenPEFile(VMPTR_PEFile vmPEFile,
                                                        DWORD &dwSize,
                                                        IStringHolder* pStrFilename)
 {
-#if !defined(FEATURE_PREJIT)
 
     return false;
```
Comment (am11) [inline src/coreclr/debug/daccess/dacdbiimpl.cpp:1254]: This method is now a dead code. Should it be deleted?

File: src/coreclr/debug/daccess/dacdbiimpl.cpp [hunk 2]
```diff
@@ -5796,42 +5742,14 @@ HRESULT DacDbiInterfaceImpl::SetNGENCompilerFlags(DWORD dwFlags)
 {
     DD_ENTER_MAY_THROW;
 
-#ifndef FEATURE_PREJIT
     return CORDBG_E_NGEN_NOT_SUPPORTED;
```
Comment (am11) [inline src/coreclr/debug/daccess/dacdbiimpl.cpp:5745]: This can also be deleted.

File: src/coreclr/debug/daccess/dacdbiimpl.cpp (hunk 1 above)
Comment (elinor-fung) [reply src/coreclr/debug/daccess/dacdbiimpl.cpp:1254]: Yes. I wanted to keep this PR to just deleting things under the `FEATURE_PREJIT` define though, so I didn't do additional cleanup. 

There's definitely a bunch of methods that now always return true/false/null or do nothing such that they and their usage can be removed. I'd rather handle those separately from the initial coarse delete of the ifdef-ed code.

**Comment by stephentoub** [comment]
@elinor-fung, should https://github.com/dotnet/runtime/blob/8ec52d68d4b39129493cdbbec0cf4f56022f6b19/src/coreclr/inc/clrconfigvalues.h#L405 and https://github.com/dotnet/runtime/blob/8ec52d68d4b39129493cdbbec0cf4f56022f6b19/src/coreclr/inc/clrconfigvalues.h#L800 have been removed as well?

**Comment by elinor-fung** [comment]
Indeed, they should: https://github.com/dotnet/runtime/pull/75911

**Pull request merged**